### PR TITLE
Embedded component

### PR DESCRIPTION
**Pull request by mhemesath**
### Summary
Pull request for embedded component. See #27 

Please add your name to the [CONTRIBUTORS.md] file. Adding your name to the [CONTRIBUTORS.md] file signifies agreement to all rights and reservations provided by the [License].

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

**Comment by mhemesath** [comment]
Need to add example/tests yet

File: packages/terra-embedded-component/src/EmbeddedComponent.jsx [hunk 1]
```diff
@@ -1,31 +1,113 @@
...
- */
-  name: PropTypes.string,
+
+ /**
+  * The basePath to load the manifest and manifest asset dependencies from.
+  * */
+  basePath: PropTypes.string.isRequired,
+
+  /**
+   * A placeholder to display while the component is laoding.
```
Comment (zhongyn) [inline packages/terra-embedded-component/src/EmbeddedComponent.jsx]: loading typo

File: packages/terra-embedded-component/docs/README.md [hunk 2]
```diff
@@ -13,9 +15,67 @@
...
 import EmbeddedComponent from 'terra-embedded-component';
 
-<EmbeddedComponent {props...} />
+<EmbeddedComponent basePath="http://mycomponent.com/path" {props...} />
 ```
 
+## Building Embeddable Components
+
+EmbeddedComponent looks for a `manifest.json` at the `basePath` to identify the component dependencies.
+The JavaScript modules are exepcted to be in [AMD](https://en.wikipedia.org/wiki/Asynchronous_module_definition) format.
```
Comment (JakeLaCombe) [inline packages/terra-embedded-component/docs/README.md]: *expected

File: packages/terra-embedded-component/docs/README.md [hunk 3]
```diff
@@ -13,9 +15,67 @@
...
+### Package the component as an AMD module
+In webpack set the following configuration for your `output` config:
+
+```js
+output: {
+  filename: '[name].[chunkhash].js',
+  path: path.join(__dirname, 'dist'),
+
+  // Required if the component is hosted from a path different than the embedding script file.
+  publicPath: 'http://componentdomain.com/',
```
Comment (JakeLaCombe) [inline packages/terra-embedded-component/docs/README.md]: Is it possible to make this match with the example you have on line 18? I think that would be easier for consumers to see how to configure this portion.

File: packages/terra-embedded-component/src/EmbeddedComponent.jsx [hunk 4]
```diff
@@ -1,31 +1,144 @@
...
 import PropTypes from 'prop-types';
-import classNames from 'classnames/bind';
-import 'terra-base/lib/baseStyles';
-import styles from './EmbeddedComponent.scss';
+import 'systemjs/dist/system';
+import { loadStylesheet } from './assets';
 
-const cx = classNames.bind(styles);
+// Expose AMD required globals on the window
+window.define = window.SystemJS.amdDefine;
```
Comment (JakeLaCombe) [inline packages/terra-embedded-component/src/EmbeddedComponent.jsx]: Should we check the value of define before we override it?

Also, could we use an expose loader for this instead potentially? https://github.com/webpack-contrib/expose-loader

File: packages/terra-embedded-component/src/EmbeddedComponent.jsx [hunk 5]
```diff
@@ -1,31 +1,144 @@
...
+  * */
+  basePath: PropTypes.string.isRequired,
+
+  /**
+   * A placeholder to display while the component is laoding.
+   * */
+  placeholder: PropTypes.element,
+
+  /**
+   * A component to display while the loading of css or module is timed out.
```
Comment (JakeLaCombe) [inline packages/terra-embedded-component/src/EmbeddedComponent.jsx]: I think this makes it a bit better

` A component to display when the css or module times out`

File: packages/terra-embedded-component/src/assets.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+/**
+* Loads a stylsheet from the given url.
```
Comment (JakeLaCombe) [inline packages/terra-embedded-component/src/assets.js]: *stylesheet

**Comment by mjhenkes** [comment]
@mhemesath mind if we close this for the time being? Site has changed since this was posted and we're leaning away from pushing in skeleton packages before the package is ready to release.

**Pull request closed without merging**